### DISCUSSION TITLE
Use ? operator on tuple member

### DISCUSSION THREAD

**Discussion by maxkoshevoi** [dotnet/csharplang#4335] [Language Ideas] [answered]
I have a list like this `List<(Func<string> name, int id)> mapping;`

Why is this syntax invalid?
` mapping.SingleOrDefault(m => m.id == 42)?.name()`
`CS0023	Operator '?' cannot be applied to operand of type '(Func<string> name, string id)'`

Could you please make it valid?

**Comment by Phantonia** [comment, marked as answer]
The tuple type `(Func<string> name, int id)` is a non-nullable value type, so accessing the `name` field is always valid and never throws. The only value that can be `null` is `name` itself (this is the case, if `SingleOrDefault` returned the default value), so you might want to use the `.name?.Invoke()` syntax, which is valid C# today.
